configure.ac

```diff
@@ -30,16 +30,38 @@ m4_define([lt_age], [3])
 m4_define([ginac_version], [ginac_major_version.ginac_minor_version.ginac_micro_version])
 m4_define([ginac_release], [ginac_major_version.ginac_minor_version])
 
-AC_INIT([pynac], ginac_version, [<pynac-devel@googlegroups.com>])
+AC_INIT([pynac], ginac_version[.sage-2020-03-28], [<pynac-devel@googlegroups.com>])
 AC_PREREQ(2.59)
 AC_CONFIG_SRCDIR(ginac/basic.cpp)
 AC_CONFIG_HEADERS(config.h)
 dnl This defines PACKAGE and VERSION.
 AM_INIT_AUTOMAKE([gnu 1.7 dist-bzip2])
 
 AM_PATH_PYTHON([2.7])
-dnl AM_PATH_PYTHON([3.4.1])
-AX_PYTHON_DEVEL
+
+dnl Adapted from CPython 3.8 cpython/Misc/python-config.in
+AC_MSG_CHECKING([for Python preprocessor flags])
+PYTHON_CPPFLAGS=$($PYTHON - <<EOD
+import sysconfig
+print(' '.join(list(('-I' + sysconfig.get_path('include'),
+                     '-I' + sysconfig.get_path('platinclude'),))
+               + sysconfig.get_config_var('CFLAGS').split()))
+EOD)
+AC_MSG_RESULT([$PYTHON_CPPFLAGS])
+AC_SUBST([PYTHON_CPPFLAGS])
+AC_MSG_CHECKING([for Python library flags])
+PYTHON_LDFLAGS=$($PYTHON - <<EOD
+import sys
+import sysconfig
+libs = list(('-lpython' + sysconfig.get_config_var('VERSION') + getattr(sys, 'abiflags', ''),))
```

Comment:
Does this mean that pynac will be linked to `-lpython`?

Reply:
If so, then this will run into https://github.com/pynac/pynac/issues/337

Reply:
Against the correct python library, yes.

Reply:
OK I will take another look at the macOS-specific linking code in distutils...

Reply:
The issue is that sometimes the python interpreter is linked to static python library and an extension linking to the shared python library will bring in 2 different python libraries to the same process which leads to segfaults. So adding `-undefined dynamic_lookup` and avoiding `-lpython` is best way forward.